asmcomp/cmmgen.ml

```diff
@@ -1346,26 +1355,23 @@ let make_switch arg cases actions dbg =
   let is_const = function
     | Cconst_int n
     | Cconst_pointer n -> (n land 1) = 1
-    | Cconst_natint _
-    | Cconst_float _
+    | Cconst_natint n
+    | Cconst_natpointer n -> (Nativeint.(to_int (logand n one) = 1))
```

Comment:
This was present in the previous version of this patch too, but anyway: what is the reason for these two tests on the tag bit?  Even if that bit was clear, the values are still constant, so I'm not sure why they can't go in the array.